### PR TITLE
Ingest via ADBC conn

### PR DESCRIPTION
**Pull request by alukach**
This PR makes use of [Arrow Database Connectivity](https://arrow.apache.org/adbc/0.3.0/index.html) to ingest data into the database.

```sh
> space2stats-ingest load postgresql://username:password@localhost:5439/postgis --parquet-file ./local.parquet
```

* load: 00:06:04
* rows/s: ~40,000

Additionally, we can also load data directly from S3.  This is somewhat slower than loading from the local filesystem.

```sh
> space2stats-ingest load postgresql://username:password@localhost:5439/postgis --parquet-file s3://wbg-geography01/Space2Stats/parquet/GLOBAL/combined_population.parquet
```

* fetch: 00:02:30
* load: 00:06:34
* rows/s: ~34,000

Before these changes, I was seeing the following ingestion times:

* load: 03:05:00
* rows/s: 1,130

### CODE REVIEW

**Comment by alukach** [comment]
Looks like tests are failing, will get to that tomorrow.

File: space2stats_api/src/space2stats_ingest/cli.py [hunk 1]
```diff
@@ -50,15 +50,17 @@ def download_and_load(
     s3_path: str,
     connection_string: str,
     local_parquet: str = typer.Option("local.parquet"),
```
Comment (zacharyDez) [inline space2stats_api/src/space2stats_ingest/cli.py:52]: We should probably keep it consistent by naming to parquet_file here to.

File: space2stats_api/src/space2stats_ingest/main.py [hunk 2]
```diff
@@ -19,7 +21,7 @@ def download_parquet_from_s3(s3_path: str, local_path: str):
 
 
 def load_parquet_to_db(
```
Comment (zacharyDez) [inline space2stats_api/src/space2stats_ingest/main.py:23]: I'm getting an error currently when the table does not yet exist:
```
space2stats-ingest load postgresql://username:password@localhost:5439/postgis --parquet-file ./local.parquet
Loading data into PostgreSQL database from ./local.parquet
Loading to PostgreSQL:   0%|                                                                                 | 0/2 [00:00<?, ?rows/s]
An error occurred: NOT_FOUND: [libpq] COPY query failed: ERROR:  relation "public.space2stats" does not exist

Query was:COPY "public" . "space2stats" ("hex_id", "sum_pop_2020", "sum_pop_f_10_2020") FROM STDIN WITH (FORMAT binary). SQLSTATE: 42P01
```

**Review by zacharyDez — CHANGES_REQUESTED** [review]

**Comment by kylebarron** [comment]
Before

> * load: 03:05:00

After

> * load: 00:06:04

That seems like a big improvement!

**Comment by kylebarron** [comment]
This is why arrow exists! It's going to be a lot faster than pandas. I think @bitner  might like this result too

File: space2stats_api/src/space2stats_ingest/main.py [hunk 3]
```diff
@@ -29,17 +31,14 @@ def load_parquet_to_db(
...
-                "space2stats", engine, if_exists="append", index=False, method="multi"
-            )
-            pbar.update(len(chunk))
+    table = pq.read_table(parquet_file)
+    with (
+        pg.connect(connection_string) as conn,
+        conn.cursor() as cur,
+        tqdm(total=table.num_rows, desc="Loading to PostgreSQL", unit="rows") as pbar,
+    ):
+        cur.adbc_ingest(TABLE_NAME, table.slice(0, 0), mode="replace")
```
Comment (zacharyDez) [inline space2stats_api/src/space2stats_ingest/main.py:40]: @alukach; replace mode creates the table if it does not exist. Slicing to 0,0 only writes the schema and then we can use efficient bulk ingesting of chunks. It also seems to have solved the failing tests.

**Review by zacharyDez — APPROVED** [review]
Awesome work @alukach! Huge gains. I'd be happy to merge this and talk later about streaming directly to the database in a future PR.

**Comment by bitner** [comment]
I suspect that the difference is not really due to arrow here. I believe that pandas `to_sql` loads data by using INSERT statements behind the scenes whereas the adbc connection is using COPY. They are both using libpq for the actual database communication, but using COPY is wayyyyy faster than INSERT, particularly if the INSERTS were separate statements rather than the multi value single statement INSERT.

File: space2stats_api/src/space2stats_ingest/main.py (hunk 3 above)
Comment (alukach) [reply space2stats_api/src/space2stats_ingest/main.py:40]: Nice! Clever with the slice.

**Pull request merged**